File: Day9/p4/searchInsert.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        if (target > nums[nums.size() - 1])
            return nums.size();
        
        vector<int> :: iterator it;
        
        it = find(nums.begin(), nums.end(), target);
        
        if (it != nums.end())
            return it - nums.begin();
        
        else{
            int i = 0;
            
            while (i < nums.size()){
                
                if (nums[i] > target)
                    return i;
                i++;
            }
        }
        return 0;
    }
};

File: Day9/p4/searchInsert.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        
        if target in nums:
            return nums.index(target)
        
        else:
            if target > nums[len(nums) -1]:
                return len(nums)           
            else:
                for i in range(len(nums)):

                    if nums[i] > target:
                        break             
        return i
